src/JetMETAnalyzer.cc

```diff
@@ -0,0 +1,517 @@
+////////////////////////////////////////////////////////////////////////////////
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+ 
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/Common/interface/RefVector.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Candidate/interface/CandidateFwd.h"
+#include "DataFormats/Candidate/interface/CandMatchMap.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/JetReco/interface/JetCollection.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/MET.h"
+#include "DataFormats/PatCandidates/interface/CompositeCandidate.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+
+#include "JetMETCorrections/Objects/interface/JetCorrector.h"
+#include "CondFormats/JetMETObjects/interface/JetCorrectorParameters.h"
+#include "CondFormats/JetMETObjects/interface/FactorizedJetCorrector.h"
+#include "PhysicsTools/Utilities/interface/LumiReWeighting.h"
+
+#include "SimDataFormats/JetMatching/interface/JetMatchedPartons.h"
+#include "SimDataFormats/GeneratorProducts/interface/GenEventInfoProduct.h"
+#include "SimDataFormats/PileupSummaryInfo/interface/PileupSummaryInfo.h"
+
+#include "JMEAnalysis/JMEValidator/interface/Analyzer.h"
+
+#include <vector>
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+////////////////////////////////////////////////////////////////////////////////
+// class definition
+////////////////////////////////////////////////////////////////////////////////
+
+class JetMETAnalyzer : public JME::Analyzer
+{
+public:
+  // construction/destruction
+  explicit JetMETAnalyzer(const edm::ParameterSet& iConfig);
+  virtual ~JetMETAnalyzer();
+
+private:
+  // member functions
+  void analyze(const edm::Event& iEvent,const edm::EventSetup& iSetup);
+  void endJob(){;}
+
+private:
+  // member data
+  std::string   JetCorLabel_;
+  std::vector<std::string> JetCorLevels_;
+
+  edm::EDGetTokenT<std::vector<pat::Jet>> srcJet_;
+  edm::EDGetTokenT<double> srcRho_;
+  edm::EDGetTokenT<std::vector<reco::Vertex>> srcVtx_;
+  edm::EDGetTokenT<std::vector<pat::Muon>> srcMuons_;
```

Comment:
Will this work with the two >> right next to each other? Shouldn't there be a space between them?